http://gitweb.freedesktop.org/?p=xorg/xserver.git;a=commit;h=876b806ec09d5ff0c6cd19df91006c4eefedfaa6
Reapply patch to fix AMD CPU detection
	author  Soren Sandmann Pedersen <ssp@dhcp83-218.boston.redhat.com>
	Tue, 6 Feb 2007 22:16:23 +0000 (17:16 -0500)
	committer  Soren Sandmann Pedersen <ssp@dhcp83-218.boston.redhat.com>
	Tue, 6 Feb 2007 22:16:23 +0000 (17:16 -0500)
	commit  876b806ec09d5ff0c6cd19df91006c4eefedfaa6
	tree  0c30b71f2adbbea5641bc4611c6360e23a2e172d  tree
	parent  13568d2aa43da4216bbcb46e1125ff28c323ac54  commit | commitdiff
Reapply patch to fix AMD CPU detection

--- xorg-server-1.2.0.orig/fb/fbpict.c	2007-03-18 21:43:21.000000000 +0100
+++ xorg-server-1.2.0/fb/fbpict.c	2007-03-19 00:30:44.000000000 +0100
@@ -1470,21 +1470,23 @@
     if (result) {
         /* result now contains the standard feature bits */
         if (result & (1 << 15))
             features |= CMOV;
         if (result & (1 << 23))
             features |= MMX;
         if (result & (1 << 25))
             features |= SSE;
         if (result & (1 << 26))
             features |= SSE2;
-        if ((result & MMX) && !(result & SSE) && (strcmp(vendor, "AuthenticAMD") == 0)) {
+        if ((features & MMX) && !(features & SSE) &&
+            (strcmp(vendor, "AuthenticAMD") == 0 ||
+             strcmp(vendor, "Geode by NSC") == 0)) {
             /* check for AMD MMX extensions */
 
             unsigned int result;            
             __asm__("push %%ebx\n"
                     "mov $0x80000000, %%eax\n"
                     "cpuid\n"
                     "xor %%edx, %%edx\n"
                     "cmp $0x1, %%eax\n"
                     "jge 2\n"
                     "mov $0x80000001, %%eax\n"
